### PR TITLE
juju/commands: Fixed incorrect delegatable-macaroon http request. Neater plans error message

### PR DESCRIPTION
**Pull request by mattyw**
By making this small change to the call to /delegatable-macaroon we will support deploying charms with terms that already have agreement from the user. Strictly speak this isn't a change that was requested for 1.25 but as the change is trivial it seems like we might as well support it.

### CODE REVIEW

**Comment by mattyw** [comment]
!!build!!

**Comment by mattyw** [comment]
!!rebuild!!

**Comment by nskaggs** [comment]
mattyw add to whitelist

**Comment by nskaggs** [comment]
!!build!!

**Comment by mattyw** [comment]
$$merge$$

File: cmd/juju/commands/deploy_test.go [hunk 1]
```diff
@@ -456,6 +456,36 @@ func (s *DeployCharmStoreSuite) TestDeployAuthorization(c *gc.C) {
...
+}
+
+func (s *stubRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if strings.HasSuffix(req.URL.Path, "/delegatable-macaroon") {
+		s.c.Check(req.URL.Query().Get("id"), gc.Equals, s.expectedCharmID)
+	}
+	return s.rt.RoundTrip(req)
+}
+
+func (s *DeployCharmStoreSuite) TestDeployDelgatableMacaroonRequest(c *gc.C) {
```
Comment (tasdomas) [inline cmd/juju/commands/deploy_test.go]: TestDeployDelegatableMacaroonRequest

**Comment by mattyw** [comment]
!!build!!

**Comment by mattyw** [comment]
!!build!!

**Pull request merged**